File: app/src/main/java/com/kakacat/minitool/translation/Contract.java
package com.kakacat.minitool.translation;

import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;

import java.util.List;

public interface Contract {

    interface View extends IView<Presenter> {
        void onAddToMyFavouriteCallBack(String s);

        void showCollectionWindow();

        void onRequestCallBack(String result, String warn);

        void showSelectLanguageWindow(int flag);
    }

    interface Presenter extends IPresenter {
        List<String> getCollectionList();

        List<String> getLanguageList1();

        List<String> getLanguageList2();

        void requestData(String input, CharSequence from, CharSequence to);

        void addToMyFavourite(String source, String target);
    }
}

File: app/src/main/java/com/kakacat/minitool/translation/Contract.kt
package com.kakacat.minitool.translation

import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView

interface Contract {
    interface View : IView<Presenter?> {
        fun onAddToMyFavouriteCallBack(s: String?)
        fun showCollectionWindow()
        fun onRequestCallBack(result: String?, warn: String?)
        fun showSelectLanguageWindow(flag: Int)
    }

    interface Presenter : IPresenter {
        val collectionList: MutableList<String>
        val languageList1: List<String>
        val languageList2: List<String>
        fun requestData(input: String?, from: CharSequence, to: CharSequence)
        fun addToMyFavourite(source: String?, target: String?)
    }
}